SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -1273,19 +1273,19 @@ void ActorSystem::ReceiveActor(Worker_EntityId EntityId)
 	ActorData& ActorComponents = ActorDataStore[EntityId];
 
 	const USpatialGDKSettings* SpatialGDKSettings = GetDefault<USpatialGDKSettings>();
-
-	AActor* EntityActor = Cast<AActor>(NetDriver->PackageMap->GetObjectFromEntityId(EntityId));
-	if (EntityActor != nullptr)
 	{
-		if (!EntityActor->IsActorReady())
+		AActor* EntityActor = Cast<AActor>(NetDriver->PackageMap->GetObjectFromEntityId(EntityId).Get(/*bEvenIfPendingKill =*/true));
```

Comment:
will this still do odd things if the actor has been killed or can that not happen

Reply:
No, we wouldn't get to this point as entities-to-be-retired won't get to this point at all.

Reply:
What is the race that this one is defending against then?

Reply:
So there are two fixes in place:

1. When an actor is destroyed for any reason, its entity ID is put into EntitiesToRetire
2. When we receive an `EntityDelta::ADD` on the AuthSubview, we'll look in EntitiesToRetire first and won't call `ReceiveActor` if the actor was destroyed
3. When we call `ReceiveActor`, we'll get the actor even if it's `PendingKill`, as opposed to before when `PendingKill` actors would be recreated. I'm not sure about how this can be repro'd, but I'd like to avoid recreating actors (unless we can come up with an expected control flow involving that). Maybe I should add `ensure(IsValid(Actor))` to solidify this assumption...

Reply:
Argh - I must've deleted the EntitiesToRetire change =_= 
The PendingKill fix is still valid.